src/Razor/src/Microsoft.VisualStudio.RazorExtension/Microsoft.VisualStudio.RazorExtension.csproj

```diff
@@ -109,12 +110,16 @@
   </PropertyGroup>
 
   <Target Name="_GenerateVSIXBindingRedirects" DependsOnTargets="PrepareForBuild;GetAssemblyVersion" BeforeTargets="CoreCompile" Inputs="$(MSBuildAllProjects)" Outputs="$(_GeneratedVSIXBindingRedirectFile)">
+    <ItemGroup>
+      <BindingRedirectAssemblies Include="@(ProjectReference)" AssemblyName="%(Filename)" />
```

Comment:
Is there any background about what this does for us and why it's needed that I can check out?

Reply:
VisualStudio is a .NET framework application and therefore needs binding redirects. So, by default we have a RazorExtension included in VisualStudio that's versioned something like 3.1.0-ABCD. When we run the RazorExtension locally we need to provide a way for VisualStudio to override what it currently has installed so we can use our locally developed bits; this is where binding redirects come in. Locally we generate binaries versioned 42.42.42.42 and then generate these binding redirects to tell VisualStudio "don't load your copy of Razor, load ours".